[BUG] Hangar Auto Publish Not Working

## Bug Description
Hangar-Auto-Publish does not work properly. It leads to a file not found error on the hangar website.

---

## Steps to Reproduce the Bug
1. Wait for build on `integration` to finish.
2. Go to the hangar website.
3. Go to Development downloads.
4. Attempt to download.

---

## Screenshots/Videos
![image](https://github.com/beanbeanjuice/SimpleProxyChat/assets/64740551/cb6001ad-5d8a-43dd-9cfd-c21140914ed4)

---

## Checklist

- [x] I have checked for similar issues.
- [x] This is a bug report, not a feature request, improvement request, or security vulnerability report.

```BASH
 FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':publishPluginPublicationToHangar' (type 'HangarPublishTask').
  - Gradle detected a problem with the following location: '/home/runner/work/SimpleProxyChat/SimpleProxyChat/build/libs/SimpleProxyChat-0.1.2.jar'.
    
    Reason: Task ':publishPluginPublicationToHangar' uses this output of task ':jar' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':jar' as an input of ':publishPluginPublicationToHangar'.
      2. Declare an explicit dependency on ':jar' from ':publishPluginPublicationToHangar' using Task#dependsOn.
      3. Declare an explicit dependency on ':jar' from ':publishPluginPublicationToHangar' using Task#mustRunAfter.
```

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -1,4 +1,5 @@
 import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+import io.papermc.hangarpublishplugin.HangarPublishTask
 import io.papermc.hangarpublishplugin.model.Platforms
 import java.io.ByteArrayOutputStream
 
@@ -176,4 +177,8 @@ tasks.withType<ShadowJar> {
 
 tasks.withType<JavaCompile> {
     options.encoding = "UTF-8"
+}
+
+tasks.withType<HangarPublishTask> {
+    dependsOn("shadowJar")
 }

